scripts/gpu-box-setup/roles/common/tasks/main.yml

```diff
@@ -0,0 +1,208 @@
+- name: Set swappiness
+  sysctl:
+    name: vm.swappiness
+    value: '{{ vm_swappiness }}'
+    state: present
+    reload: yes
+
+- name: Optimize transparent huge pages for memory-intensive workloads
+  sysctl:
+    name: vm.nr_hugepages
+    value: '{{ vm_nr_hugepages }}'
+    state: present
+    reload: yes
+  when: optimize_for_memory | bool
+
+- name: Optimize networking for GPU workloads
+  sysctl:
+    name: "{{ item.key }}"
+    value: "{{ item.value }}"
+    state: present
+    reload: yes
+  loop:
+    - { key: 'net.core.rmem_max', value: '{{ tcp_rmem_max }}' }
+    - { key: 'net.core.wmem_max', value: '{{ tcp_wmem_max }}' }
+    - { key: 'net.ipv4.tcp_rmem', value: '4096 87380 {{ tcp_rmem_max }}' }
+    - { key: 'net.ipv4.tcp_wmem', value: '4096 65536 {{ tcp_wmem_max }}' }
+
+# Configure swap for AI workloads if needed
+- name: Check if swap file exists
+  stat:
+    path: /swapfile
+  register: swap_file_check
+
+- name: Create swap file if it doesn't exist and swap_size is defined
+  command: fallocate -l {{ swap_size }}G /swapfile
+  args:
+    creates: /swapfile
+  when: 
+    - not swap_file_check.stat.exists
+    - swap_size is defined and swap_size > 0
+
+- name: Set swap file permissions
+  file:
+    path: /swapfile
+    owner: root
+    group: root
+    mode: '0600'
+  when: 
+    - swap_file_check.stat is defined and swap_file_check.stat.exists
+    - swap_size is defined and swap_size > 0
+
+- name: Set up swap area
+  command: mkswap /swapfile
+  args:
+    creates: /proc/swaps
+  when: 
+    - swap_file_check.stat is defined and swap_file_check.stat.exists
+    - swap_size is defined and swap_size > 0
+  register: mkswap_result
+
+- name: Enable swap
+  command: swapon /swapfile
+  when: 
+    - swap_file_check.stat is defined and swap_file_check.stat.exists
+    - swap_size is defined and swap_size > 0
+    - mkswap_result is defined and mkswap_result.changed
+
+- name: Add swap to fstab
+  lineinfile:
+    path: /etc/fstab
+    line: "/swapfile none swap sw 0 0"
+    state: present
+  when: 
+    - swap_file_check.stat is defined and swap_file_check.stat.exists
+    - swap_size is defined and swap_size > 0
+
+# Install system monitoring tools
+- name: Install system monitoring tools
+  package:
+    name: "{{ monitoring_packages }}"
+    state: present
+  when: install_monitoring | bool
+
+# Set up Python virtual environment for AI development
+- name: Create Python virtual environment directory
+  file:
+    path: "{{ python_venv_path }}"
+    state: directory
+    mode: '0755'
+  when: install_ai_dependencies | bool
+
+- name: Create Python virtual environment
+  command: "python{{ python_version }} -m venv {{ python_venv_path }}"
+  args:
+    creates: "{{ python_venv_path }}/bin/activate"
+  when: install_ai_dependencies | bool
+
+- name: Upgrade pip in virtual environment
+  pip:
+    name: pip
+    state: latest
+    virtualenv: "{{ python_venv_path }}"
+  when: install_ai_dependencies | bool
```

Comment:
Can we use `uv` you think?

Reply:
these are to be in the global scope. shall i use uv there too? WDYT?

Reply:
I think we should not need to do any ai dependencies in a global scope at all. I think all the packages we would need for various things should be in an virtual environment. What is the purpose of the global packages?

Reply:
it allows easier fast scripting experimnetation, without needing to create a project-structure.